spec/features/medicaid_spec.rb

```diff
@@ -1,8 +1,142 @@
 require "rails_helper"
 
 RSpec.feature "Medicaid app" do
+  scenario "with a single member", :js do
+    visit medicaid_root_path
+
+    within(".slab--hero") do
+      click_on "Apply for Medicaid"
+    end
+
+    on_pages "Introduction" do
+      click_on "Yes"
+
+      fill_in "What is your first name?", with: "Jessie"
+      fill_in "What is your last name?", with: "Tester"
+      select_radio(question: "What is your gender?", answer: "Female")
+      click_on "Next"
+
+      expect(page).to have_content(
+        "Now tell us about any other people residing in your household.",
+      )
+      click_on "Next"
+
+      expect(page).to have_content("Are you currently a college student?")
+      click_on "No"
+
+      expect(page).to have_content("Are you currently a US Citizen?")
+      click_on "Yes"
+    end
+
+    on_pages "Health Coverage Needs" do
+      expect(page).to have_content(
+        "Are you currently enrolled in a health insurance plan?",
+      )
+      click_on "No"
+
+      expect(page).to have_content(
+        "Do you need help paying for medical expenses from the last 3 months?",
+      )
+      click_on "Yes"
+    end
+
+    on_pages "Quick Health Questions" do
+      expect(page).to have_content("Do you have a disability?")
+      click_on "No"
+
+      expect(page).to have_content("Have you been pregnant recently?")
+      click_on "No"
+    end
+
+    on_pages "Quick Tax Question" do
+      expect(page).to have_content(
+        "Do you plan on filing a federal tax return next year?",
+      )
+
+      click_on "Yes"
+    end
+
+    on_pages "Current Income" do
+      expect(page).to have_content("Do you currently have a job?")
+      click_on "Yes"
+
+      expect(page).to have_content("Tell us how many jobs you currently have.")
+      choose "1 job"
+      click_on "Next"
+
+      expect(page).to have_content("Are you self-employed?")
+      click_on "No"
+
+      expect(page).to have_content("Do you get income that's not from a job?")
+      click_on "Yes"
+
+      expect(page).to have_content(
+        "What type of income do you receive that’s not from a job?",
+      )
+      check "Unemployment"
+      click_on "Next"
+    end
+
+    on_pages "Current Expenses" do
+      expect(page).to have_content(
+        "Do you pay child support, alimony or arrears?",
+      )
+      click_on "No"
+
+      expect(page).to have_content("Do you pay student loan interest?")
+      click_on "No"
+    end
+
+    on_pages "Income & Expense Amounts" do
+      click_on "Next"
+
+      fill_in "Your Current Job", with: 100
+      fill_in "Your Unemployment", with: 100
+      click_on "Next"
+    end
+
+    on_pages "Contact Information & Followup" do
```

Comment:
we usually call `on_page` for each new page because that method checks that you are on the correct page, which is good for debugging. If we call around a group is it doing that for each visited page?

Reply:
It's not doing it for *each* page. But since these pages' titles aren't ever going to change I thought it would be sufficient to group them. If we want to test *each* page then a view spec with some sort of terse, concise test on the title might be a good idea.

As a run through these pages, however, getting a feel for the interaction seems to be the most important part. Making sure the clicks take us to where they should is the most important part. IMHO.

What do you think?

Reply:
Meh, I am ok with either way. If it becomes a problem we can always add later